src/extensions/locale/ext-locale-init.reb

```diff
@@ -466,13 +466,13 @@ unless 'Windows = first system/platform [
         letter: charset [#"a" - #"z" #"A" - #"Z"]
         unless parse env-lang [
             copy lang: [some letter]
-            #"_" copy territory: [some letter]
-            to end
+            [#"_" copy territory: [some letter] | #"." (territory: copy "US")]
+            to end (if lang = "C" [lang: copy "en"])
         ][
-            fail spaced ["Malformated env LANG:" env-lang]
+            fail spaced ["Non-standard env LANG:" env-lang]
         ]
```

Comment:
Like you said in the chat room, it might be better to print a warning instead of failing. Also, I think a simple "C" is a valid LANG setting. According to [this](http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html), the format is:
```
[language[_territory][.codeset][@modifier]]
```